### PR TITLE
Refactor handle-inter-router-request-fn

### PR DESCRIPTION
**Pull request by sradack**
## Changes proposed in this PR

Refactor `handle-inter-router-request-fn` in middleware style

## Why are we making these changes?

This PR simplifies the code for handling inter-router authentication.

- Brings us in line with intended use pattern of `basic-authentication/wrap-basic-authentication`
- Prefers using request map directly in handlers to get `src-router-id`, instead of extracting data from request upstream and passing separately as parameters.
- Removes parameters from anonymous function that didn't need to be curried.

### CODE REVIEW

File: waiter/src/waiter/core.clj [hunk 1]
```diff
@@ -1304,8 +1276,21 @@
...
-                                   request)))})
+                                      wrap-router-auth-fn]
+                               (wrap-router-auth-fn
+                                 (fn [request]
+                                   (handler/work-stealing-handler instance-rpc-chan request))))
+   :wrap-router-auth-fn (pc/fnk [[:state passwords router-id]]
+                          (fn wrap-router-auth-fn [handler]
+                            (fn [request]
+                              (let [router-comm-authenticated?
+                                  (fn router-comm-authenticated? [source-id secret-word]
```
Comment (shamsimam) [inline waiter/src/waiter/core.clj]: Indentation looks off.

File: waiter/src/waiter/handler.clj [hunk 2]
```diff
@@ -67,7 +67,7 @@
 
 (defn complete-async-handler
   "Completes execution of an async request by propagating a termination message to the request monitor system."
-  [async-request-terminate-fn src-router-id {:keys [route-params uri] :as request}]
+  [async-request-terminate-fn {:keys [route-params uri] {:keys [src-router-id]} :basic-authentication :as request}]
```
Comment (shamsimam) [inline waiter/src/waiter/handler.clj]: Should we consider a utility function: `request->source-router-id`, the destructuring feels unwieldy.

File: waiter/test/waiter/scaling_test.clj [hunk 3]
```diff
@@ -151,7 +151,7 @@
                       correlation-id
                       (kill-instance-handler
                         scheduler instance-rpc-chan inter-kill-request-wait-time-ms blacklist-config peers-acknowledged-blacklist-requests-fn
-                        src-router-id {:route-params {:service-id test-service-id}}))
+                        {:route-params {:service-id test-service-id} :basic-authentication {:src-router-id src-router-id}}))
```
Comment (shamsimam) [inline waiter/test/waiter/scaling_test.clj]: Nitpick: lexicographic ordering

File: waiter/test/waiter/scaling_test.clj [hunk 4]
```diff
@@ -175,7 +175,7 @@
                     correlation-id
                     (kill-instance-handler
                       scheduler instance-rpc-chan inter-kill-request-wait-time-ms blacklist-config peers-acknowledged-blacklist-requests-fn
-                      src-router-id {:route-params {:service-id test-service-id}}))
+                      {:route-params {:service-id test-service-id} :basic-authentication {:src-router-id src-router-id}}))
```
Comment (shamsimam) [inline waiter/test/waiter/scaling_test.clj]: Nitpick: lexicographic ordering

File: waiter/test/waiter/scaling_test.clj [hunk 5]
```diff
@@ -203,7 +203,7 @@
...
```
Comment (shamsimam) [inline waiter/test/waiter/scaling_test.clj]: Nitpick: lexicographic ordering

File: waiter/src/waiter/handler.clj (hunk 2 above)
Comment (sradack) [reply waiter/src/waiter/handler.clj]: I'd prefer to leave as is

File: waiter/src/waiter/core.clj (hunk 1 above)
Comment (sradack) [reply waiter/src/waiter/core.clj]: Fixed

File: waiter/test/waiter/scaling_test.clj (hunk 3 above)
Comment (sradack) [reply waiter/test/waiter/scaling_test.clj]: Fixed

File: waiter/test/waiter/scaling_test.clj (hunk 4 above)
Comment (sradack) [reply waiter/test/waiter/scaling_test.clj]: Fixed

File: waiter/test/waiter/scaling_test.clj (hunk 5 above)
Comment (sradack) [reply waiter/test/waiter/scaling_test.clj]: Fixed

**Comment by sradack** [comment]
@shamsimam all feedback addressed

**Pull request merged**